libnmstate/netapplier.py

```diff
@@ -388,11 +407,38 @@ def _remove_iface_ipv6_link_local_addr(desired_state, current_state):
     return desired_state, current_state
 
 
+def _remove_iface_dhcp_manual_addr(desired_state, current_state):
+    for state in (desired_state, current_state):
+        for family in ('ipv4', 'ipv6'):
+            if state[family]['dhcp']:
+                state[family]['address'] = []
+    return desired_state, current_state
+
+
+def _canonicalize_ipv4_state(desired_state, current_state):
+    base_conf = {
+        'ipv4': {
+            'enabled': False,
+            'address': [],
+            'dhcp': False,
+        }
+    }
+    desired_state = _dict_update(copy.deepcopy(base_conf), desired_state)
```

Comment:
Why do we need to do this? For the desired state you have already done this in the `sanitize_dhcp_state` func.
Can't we do the same here for the current state?
Feels like `_remove_iface_dhcp_manual_addr` is also in this game.

Same for the IPv6 one.